dna/zomes/courses/code/src/course.rs

```diff
@@ -259,15 +454,86 @@ pub fn add_module_to_course(
     course_address: &Address,
     module_address: &Address,
 ) -> ZomeApiResult<Address> {
-    let current_course = hdk::get_entry(course_address).unwrap().unwrap();
-    if let Entry::App(_, current_course) = current_course {
-        let mut course_entry = Course::try_from(current_course.clone())
+    let current_course_anchor = hdk::get_entry(course_address).unwrap().unwrap();
```

Comment:
Much better to use option's `.ok()` and so on functions than unsafely unwrapping. This can lead to panics and we don't ever want those.